app/controllers/users_controller.rb

```diff
@@ -3,6 +3,27 @@ class UsersController < ApplicationController
   before_action :retrieve_patients, only: [:add_patient, :remove_patient]
   rescue_from Mongoid::Errors::DocumentNotFound, with: -> { head :bad_request }
 
+  def create
+    raise 'Permission Denied' unless current_user.admin?
```

Comment:
I think you're right about this. I might wind up wrapping that in its own error class, but I think you're right on.